src/vma/iomux/epfd_info.cpp

```diff
@@ -290,16 +290,21 @@ int epfd_info::add_fd(int fd, epoll_event *event)
 			          event->events & ~SUPPORTED_EPOLL_EVENTS);
 			m_log_invalid_events--;
 		}
+
+		if ((socket_epfd = temp_sock_fd_api->get_epoll_context_fd())) {
+			if (socket_epfd == m_epfd) {
```

Comment:
@liranoz12, I agree with the correctness of this logic, but I think it is better if it is moved into `sockinfo::add_epoll_context()`. So that add/remove_epoll_context() will return error code which can be verified and returned to user as correct errno.
In this case you don't need to add `int get_epoll_context_fd()`

The reason for this change is due to the source of the limitation, which comes from logic we have in sockinfo. If we decided to expand the epfd array in sockinfo, this code in epfd_info will suddenly be wrong.